Lib/tkinter/__init__.py

```diff
@@ -1385,11 +1385,14 @@ def bind(self, sequence=None, func=None, add=None):
         return self._bind(('bind', self._w), sequence, func, add)
 
     def unbind(self, sequence, funcid=None):
-        """Unbind for this widget for event SEQUENCE  the
-        function identified with FUNCID."""
-        self.tk.call('bind', self._w, sequence, '')
+        """Unbind for this widget the event SEQUENCE. if
+        FUNCID is given, delete the command also."""
+        bound = ''
         if funcid:
             self.deletecommand(funcid)
+            funcs = self.tk.call('bind', self._w, sequence, None).split('\n')
+            bound = '\n'.join([f for f in funcs if not f.startswith(f'if {{"[{funcid}')])
```

Comment:
`join` does not need the intermediate list and the second iteration.  Remove '[' and ']' to leave a generator expression (comprehension).

I don't understand `f.startswith(f'if {{"[{funcid}')`, ie, why the func corresponding to funcid should necessarily be a string starting with something like `if {"[<funcid>`.  Does this have something to do with how tcl wraps python functions?  If so, should we assume that the bound function is such?

Reply:
You'right for the generator. 
Neither do I. After having done several tests to check how callbacks are bound, it came out that they always start with that pattern, so I assumed this is "how tcl wraps python functions".